### ISSUE TITLE
Consider adding a BodyWritable[Json]

### ISSUE THREAD

**Issue by tbinna** [jilen/play-circe#36]
With the update to Play 2.6 my Play WS clients broke.

Previously body serialization when calling `put`, `post`, etc. worked using `play.api.http.Writable` and type class instances `writableOf_Json` in Circe.scala did the job.

Now play-ws requires a type class `play.api.libs.ws.BodyWritable` which is not included in this project.

For now I defined this trait in my project which does the job:

```
trait CirceJsonBodyWritables {

  val defaultPrinter = Printer.noSpaces

  implicit def bodyWritableOf_Json(
      implicit printer: Printer = defaultPrinter): BodyWritable[Json] = {
    BodyWritable(
      json => InMemoryBody(ByteString.fromString(json.pretty(printer))),
      "application/json")
  }

}
```

Would be nice to add this to the Circe.scala trait but it requires you to add `play-ws` dependency.

**Comment by jilen** [comment]
@tbinna  `Circe` trait is basically for `Controller`, `BodyWritable` instances seems should be available somewhere else

**Comment by tbinna** [comment]
Yes, that's true. It could be a different trait for Play WS. For Play 2.5.x I used the same `Circe` trait for my WS clients. This worked because play-ws was using the same `Writable` to serialize bodies. Now with Play 2.6 I still use the `Circe` trait but I need to define a `BodyWritable[io.circe.Json]` for WS clients.
If I find some time I will create PR, then we can discuss based on that.

**Comment by romanskie** [comment]
Hey Guys,

so is there by now a way or workaround to parse WSReponse bodys to case classes? I tried to write a custom CirceJsonBodyReadables trait by myself, but have to admit that I am not experienced enough. Maybe I am wrong, but I cant just use @tbinna writable trait for that right? I would like to help integrating that in play-circe!

**Comment by jilen** [comment]
@romanskie  I think it would be better not casting to `StandaloneAhcWSResponse` 
but relies on `WSResponse`.

**Comment by jilen** [comment]
@romanskie  Also you may define 
```scala
implicit def circeBodyReadable[A: io.circe.Decoder] = BodyReadable[io.circe.Decoder.Result[A]]] { response =>
   decode[response.body]
}
```
Which will enable parse response to `Decoder.Result[A]`

**Comment by romanskie** [comment]
I wrote this now:
`
 implicit def circeJsonBodyReadable = BodyReadable[io.circe.Json] { response =>
    io.circe.parser.parse(response.bodyAsBytes.utf8String) match {
      case Left(decodingFailure) => throw decodingFailure
      case Right(json) => json
    }
  }
`

And now I am able to:
 
`
wsClient.url(btcURL).get().map(response => response.body[Json].as[CryptoCompareResponse])
`
That was the only solution I could apply. It would be very nice if I just could use the following:

`
wsClient.url(btcURL).get().map(response => response.body[CryptoCompareResponse])
`
I guess that is what you wanted me to do with your implicit def, but i could not get it to work!
